src/createAnimatedComponent.tsx

```diff
@@ -133,6 +134,8 @@ export type AnimatedComponentProps<P extends Record<string, unknown>> = P & {
     | typeof BaseAnimationBuilder
     | EntryExitAnimationFunction
     | Keyframe;
+  sharedTransitionTag?: string;
+  sharedTransitionStyle?: any;
```

Comment:
would 
```suggestion
  sharedTransitionStyle?:
    | ILayoutAnimationBuilder
    | LayoutAnimationFunction
    | Keyframe;
```
work?

Reply:
Resolved in: https://github.com/software-mansion/react-native-reanimated/pull/4014/commits/d28696f8d202447293a16ad73004c52c1b2b48d3